### PR TITLE
Use f-strings for string formatting rather than .format(...).

### PR DESCRIPTION
**Pull request by aslanides**
This should tidy up/improve readability somewhat.

### CODE REVIEW

File: jax/api.py [hunk 1]
```diff
@@ -1198,12 +1197,12 @@ def vmap(fun: F, in_axes=0, out_axes=0, axis_name=None) -> F:
...
-           "those types as leaves, but got {}.")
-    raise TypeError(msg.format(in_axes))
+           f"those types as leaves, but got {in_axes}.")
+    raise TypeError(msg)
   if not all(isinstance(l, (type(None), int)) for l in out_axes_):
     msg = ("vmap out_axes must be an int, None, or (nested) container with "
-           "those types as leaves, but got {}.")
-    raise TypeError(msg.format(out_axes))
+           f"those types as leaves, but got {out_axes}.")
+    raise TypeError(msg)
```
Comment (jakevdp) [inline jax/api.py:1205]: nit: remove the `msg` variable here, as you did elsewhere?

**Review by jakevdp — COMMENTED** [review]
Looks good, this is something that I've been meaning to do some time :grin:

Only comment: there are still a few places that use the pattern `msg = "..."; raise TypeError(msg)`. I have a slight preference toward avoiding the `msg` variable and passing the f-string literal to the TypeError. What do you think?

File: jax/api.py (hunk 1 above)
Comment (aslanides) [reply jax/api.py]: Done.

**Comment by aslanides** [comment]
Thanks Jake! PTAL.

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Thanks!

**Pull request merged**